usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlockTest.java

```diff
@@ -1615,37 +1609,427 @@ void tripWithNullLastTimeShouldNotGenerateNotice() {
 
-        verify(secondTripFirstStopTime, times(2)).getArrivalTime();
-        verify(secondTripLastStopTime, times(2)).getDepartureTime();
-        verify(secondTripLastStopTime, times(2)).getDepartureTime();
+        verify(secondTripLastStopTime, times(3)).getDepartureTime();
 
         verify(thirdTripLastStopTime, times(4)).getDepartureTime();
 
-        verify(mockTimeUtils, times(1))
-                .arePeriodsOverlapping(ArgumentMatchers.anyInt(), ArgumentMatchers.anyInt(),
-                        ArgumentMatchers.anyInt(), ArgumentMatchers.anyInt());
+        verifyNoInteractions(mockResultRepo, mockTimeUtils);
+        verifyNoMoreInteractions(mockLogger, mockDataRepo, firstMockTrip, secondMockTrip, thirdMockTrip,
+                firstTripFirstStopTime, firstTripSecondStopTime, firstTripThirdStopTime, firstTripLastStopTime,
+                secondTripFirstStopTime, secondTripLastStopTime, thirdTripFirstStopTime, thirdTripLastStopTime,
+                firstCalendarDateForServiceA, secondCalendarDateForServiceA, calendarDateForServiceB,
+                calendarDateForServiceC);
+    }
+
+    @Test
+    void nonOverlappingTripsWithDifferentServiceIdShouldNotGenerateNoticeWhenComparingTripsReferencingBothFiles() {
+        // trips.txt
+        // | routeId | tripId | serviceId  | blockId |
+        // |---------|--------|----------- |---------|
+        // | 0       | 2      | a          | 7       |
+        // | 0       | 5      | b          | 7       |
+        // | 0       | 8      | c          | 7       |
+        // | 1       | 9      | d          | 7       |
+
+        // stop_times.txt
+        // | tripId | arrivalTime | departureTime | stopId | stopSequence  |
+        // |--------|-------------|---------------|--------|---------------|
+        // | 2      | 05:00       | 05:00         |  99    |  8            |
+        // | 2      | 06:00       | 06:30         | 100    |  9            |
+        // | 5      | 07:00       | 07:40         | 101    | 10            |
+        // | 5      | 08:20       | 10:00         | 102    | 11            |
+        // | 8      | 11:03       | 11:15         | 103    | 12            |
+        // | 8      | 11:16       | 11:20         | 104    | 13            |
+        // | 9      | 07:00       | 07:40         | 105    | 14            |
+        // | 9      | 07:40       | 07:59         | 106    | 15            |
+
+        // trips.txt + stop_times.txt
+        // | routeId | tripId | serviceId | blockId | first stop time | last stop time |
+        // |---------|--------|-----------|---------|-----------------|----------------|
+        // | 0       | 2      | a         | 7       | 05:00           | 06:30          |
+        // | 0       | 5      | b         | 7       | 07:00           | 10:00          |
+        // | 0       | 8      | c         | 7       | 11:03           | 11:20          |
+        // | 1       | 9      | c         | 7       | 07:00           | 10:00          |
```

Comment:
I think this should be:

```suggestion
        // | 0       | 9      | d         | 7       | 06:00           | 10:00          |
```

That means that "normally" you'd run one long trip from 6am to 10am Thurs, Fri, and Sat.

Based on my other comments, the only exception would be 20200820 (service `c`), which instead runs only from 11:03 to 11:20. And then for service `a` we also have a trip running on 20200723 and 20200801 from 5:00 to 6:30 which shouldn't overlap due to the dates.

If you set it up this way, then for the other test `overlappingTripsWithDifferentServiceIdShouldGenerateNoticeWhenComparingTripsReferencingBothFiles()` where you want to detect overlap you can change the service `a` dates to 20200827 and 20200828 in calendar_dates.txt. That should cause an overlap on Aug 27 and Aug 28 between 6 and 6:30.